addon/validations/factory.js

```diff
@@ -207,21 +207,21 @@ function createValidationsClass(inheritedValidationsClass, validations = {}) {
     return obj;
   }, validationRules);
 
-  // Create the CPs that will be a part of the `attrs` object
-  const attrCPs = validatableAttributes.reduce((obj, attribute) => {
-    assign(obj, attribute, createCPValidationFor(attribute, get(validationRules, attribute)), true);
-    return obj;
-  }, {});
-
   // Create the mixin that holds all the top level validation props (isValid, messages, etc)
   const TopLevelProps = createTopLevelPropsMixin(validatableAttributes);
 
   // Create the `attrs` class which will add the current model reference once instantiated
-  const Attrs = Ember.Object.extend(attrCPs, {
+  const Attrs = Ember.Object.extend({
     init() {
       this._super(...arguments);
       const model = this.get('_model');
 
+      // Create the CPs
+      validatableAttributes.forEach(attribute => {
+        const cp = createCPValidationFor(attribute, get(validationRules, attribute), owner);
+        assign(this, attribute, cp, true);
```

Comment:
This is a feature that only exists due to a bug in ember, lets not rely on it. Basically there is no way currently to performently add CP's to instances, only classes. I have toyed around with the idea of adding support for adding instance specific dependentKeys to existing CP's, but it hasn't be implemented yet. Would that help?